### PR TITLE
Small e-mail fixes

### PR DESCRIPTION
**Pull request by mitar**
Just some things I noticed while reading through the code.

Not tested.

### CODE REVIEW

**Comment by garply** [comment]
Hey admins! Can one of you let me know if it's safe to build this pull request? See https://alpha.sandstorm.io/grain/kB7bofATTL2jGuKSFwmqdA/GithubPRCommands for a list of commands.

**Comment by kentonv** [comment]
Hmm, it's clear the code does not match the comments, but I wonder if it is the code or the comments that is wrong. Maybe we actually intended to require that the grain specify an address matching one of the two, without any default? @jparyani do you remember?

(The unrelated `inReplyTo` change is clearly a bug, of course.)

**Comment by mitar** [comment]
In documentation it says:

> When you send e-mail from a Sandstorm app, we allow you to set the "From" header either to your app's random address or to your verified Sandstorm login address. 

It  does not "feel" like you have to set it, but you are allowed to set it?

**Comment by kentonv** [comment]
Yeah that line is really ambiguous, but the documentation was meant to be descriptive, not prescriptive, so that doesn't help us figure out what was the original intended behavior here...

**Comment by mitar** [comment]
How does app figure out what is its valid outgoing address? For example, how can a Meteor app know that? Is this easy? Then we can require it to set the right from address.

**Comment by kentonv** [comment]
Apps normally call `HackSessionContext#getPublicId()` and then form the address as publicId@hostname.

https://github.com/sandstorm-io/sandstorm/blob/master/src/sandstorm/hack-session.capnp#L35

**Comment by kentonv** [comment]
Or to get the user's address, `HackSessionContext#getUserAddress()`:

https://github.com/sandstorm-io/sandstorm/blob/master/src/sandstorm/hack-session.capnp#L63

Keep in mind `HackSessionContext` is a born-deprecated API.

**Comment by mitar** [comment]
Hm, and that is not available for Meteor apps, no? Is it even possible to send e-mails from Meteor apps?

**Comment by kentonv** [comment]
You can use node-capnp to make Cap'n Proto requests from Meteor apps.

The documentation describes how to get at HackSessionContext: https://docs.sandstorm.io/en/latest/developing/email-from-apps/

File: shell/server/drivers/mail.js [hunk 1]
```diff
@@ -216,15 +216,17 @@ hackSendEmail = (session, email) => {
...
-      email.from = {};
+      email.from = {
+        address: grainAddress
+      };
     }
 
-    const grainAddress = session._getAddress();
-    const userAddress = session._getUserAddress();
-
-    // First check if we're changing the from address, and if so, move it to reply-to
```
Comment (mitar) [inline shell/server/drivers/mail.js]: I also removed this comment. It is not true at all.

**Comment by mitar** [comment]
There could be a Meteor package for that. ;-)

**Comment by kentonv** [comment]
Thanks!

**Pull request merged**